diff --git a/modules/hosting/task.hosting.inc b/modules/hosting/task.hosting.inc
index 31dd56f..6c95989 100644
--- a/modules/hosting/task.hosting.inc
+++ b/modules/hosting/task.hosting.inc
@@ -128,7 +128,7 @@ function drush_hosting_task() {
  */
 function drush_hosting_hosting_task_rollback() {
   $task =& drush_get_context('HOSTING_TASK');
-  module_invoke_all(sprintf("hosting_%s_task_rollback", $task->task_type), $task, drush_get_context('HOSTING_DRUSH_OUTPUT'));
+  module_invoke_all(sprintf("hosting_%s_task_rollback", str_replace('-', '_', $task->task_type)), $task, drush_get_context('HOSTING_DRUSH_OUTPUT'));
 }
 
 /**
@@ -139,5 +139,5 @@ function drush_hosting_hosting_task_rollback() {
 function drush_hosting_post_hosting_task($task) {
   $task =& drush_get_context('HOSTING_TASK');
 
-  module_invoke_all(sprintf("post_hosting_%s_task", $task->task_type), $task, drush_get_context('HOSTING_DRUSH_OUTPUT'));
+  module_invoke_all(sprintf("post_hosting_%s_task", str_replace('-', '_', $task->task_type)), $task, drush_get_context('HOSTING_DRUSH_OUTPUT'));
 }
